rpc/src/v1/types/transaction.rs

```diff
@@ -51,6 +51,9 @@ pub struct Transaction {
 	pub creates: Option<H160>,
 	/// Raw transaction data
 	pub raw: Bytes,
+	/// Public key of the signer.
+	#[serde(rename="publicKey")]
+	pub public_key: Option<H512>,
```

Comment:
other thing to note is that we already provide a non-standard block structure (with the key `sealFields`).